### DISCUSSION TITLE
Persist filters/facetes data from modal to component

### DISCUSSION THREAD

**Discussion by Ekeu** [algolia/instantsearch#5286] [Q&A] [answered]
Hello Algolia 

I totally agree my title isn't pretty clear. But to detail out my problem, I'm actually working with react-instantsearch and I successfully added all the required widgets (filters etc...)  I needed. But then, when trying to work around with the mobile view, I decided to put all my filter in a modal that opens only when the user clicks on the filter button.

The issue I'm facing is that when the user selects all his filters in the modal and clicks on the button to see the products when the modal disappears, the main component resets and the chosen filters aren't applied. Here is a video showing what I'm talking about:

https://user-images.githubusercontent.com/23312714/121373093-16f1cc80-c93f-11eb-81ab-1b93ed9a1517.mp4

I tried implementing using the same idea from [here](https://codepen.io/e-krebs/pen/RwwvgeJ?editors=0000) but  I couldn't because I'm using a modal that constantly mounts and unmounts.

Here is my component:

![code-1](https://user-images.githubusercontent.com/23312714/121375006-ac419080-c940-11eb-98ca-c884e7af01c4.png)

I'll really appreciate if you guys could give me some ideas on how to implement that.

Thanks and have a nice day

**Comment by Haroenv** [comment, marked as answer]
This should work out of the box, although you need to do two things:

1. make it so that the slide over hides (with css) the invisible content, instead of unmounting it
2. make the slide over a child of the InstantSearch component
